lisa/tools/amdsmi.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import re
+
+from lisa.executable import Tool
+from lisa.util import LisaException, find_groups_in_lines
+
+
+class AmdSmi(Tool):
+    # Pattern to match GPU entries in amd-smi list output
+    # Example:
+    # GPU: 0
+    #     BDF: N/A
+    #     UUID: N/A
+    #     KFD_ID: 65402
+    #     NODE_ID: 2
+    #     PARTITION_ID: 0
+    _gpu_pattern = re.compile(r"^GPU:\s+\d+", re.MULTILINE)
+
+    @property
+    def command(self) -> str:
+        return "amd-smi"
+
+    @property
+    def can_install(self) -> bool:
+        return False
+
+    def get_gpu_count(self) -> int:
+        """
+        Get the number of AMD GPUs detected by amd-smi.
+        Uses 'amd-smi list' command which shows all GPU devices.
+        """
+        result = self.run("list")
+        if result.exit_code != 0 or (result.exit_code == 0 and result.stdout == ""):
+            result = self.run("list", sudo=True)
+            if result.exit_code != 0 or (result.exit_code == 0 and result.stdout == ""):
+                raise LisaException(
+                    f"Failed to query AMD GPUs using 'amd-smi list'. "
+                    f"Exit code: {result.exit_code}, "
+                    f"stdout: '{result.stdout}', "
+                    f"stderr: '{result.stderr}'. "
```

Comment:
Print stderr first, because the string is truncated in database.

Reply:
updated